hudi-common/src/main/java/org/apache/hudi/common/util/JsonUtils.java

```diff
@@ -35,6 +36,8 @@ public class JsonUtils {
   private static final ObjectMapper MAPPER = new ObjectMapper();
 
   static {
+    registerModules(MAPPER);
+
```

Comment:
This change is ported from #8840 . I assume we will need to merge that PR before this one?

Reply:
#8840 contains some minor improvements which I'm currently not inclined to include.  I'll revise #8840 to contain necessary fixes and then land it before this PR.

Reply:
This is fixed now.